src/main/java/org/gridsuite/modification/server/entities/equipment/creation/TwoWindingsTransformerCreationEntity.java

```diff
@@ -69,14 +156,112 @@ public TwoWindingsTransformerCreationEntity(String equipmentId,
         this.magnetizingSusceptance = magnetizingSusceptance;
         this.ratedVoltage1 = ratedVoltage1;
         this.ratedVoltage2 = ratedVoltage2;
+        this.ratedS = ratedS;
+
+        this.phaseTapChangerLowTapPosition = phaseTapChangerLowTapPosition;
+        this.phaseTapChangerTapPosition = phaseTapChangerTapPosition;
+        this.phaseTapChangerRegulating = phaseTapChangerRegulating;
+        this.phaseTapChangerTargetDeadband = phaseTapChangerTargetDeadband;
+        this.phaseTapChangerTerminalRefConnectableId = phaseTapChangerTerminalRefConnectableId;
+        this.phaseTapChangerTerminalRefVoltageLevelId = phaseTapChangerTerminalRefVoltageLevelId;
+        this.phaseTapChangerTerminalRefType = phaseTapChangerTerminalRefType;
+        this.phaseTapChangerRegulationMode = phaseTapChangerRegulationMode;
+        this.phaseTapChangerRegulationValue = phaseTapChangerRegulationValue;
+        this.ratioTapChangerLowTapPosition = ratioTapChangerLowTapPosition;
+        this.ratioTapChangerTapPosition = ratioTapChangerTapPosition;
+        this.ratioTapChangerRegulating = ratioTapChangerRegulating;
+        this.ratioTapChangerTargetDeadband = ratioTapChangerTargetDeadband;
+        this.ratioTapChangerTerminalRefConnectableId = ratioTapChangerTerminalRefConnectableId;
+        this.ratioTapChangerTerminalRefVoltageLevelId = ratioTapChangerTerminalRefVoltageLevelId;
+        this.ratioTapChangerTerminalRefType = ratioTapChangerTerminalRefType;
+        this.ratioTapChangerLoadTapChangingCapabilities = ratioTapChangerLoadTapChangingCapabilities;
+        this.ratioTapChangerTargetV = ratioTapChangerTargetV;
+        this.tapChangerSteps = tapChangerSteps;
     }
 
     @Override
     public TwoWindingsTransformerCreationInfos toModificationInfos() {
         return toTwoWindingsTransformerCreationInfosBuilder().build();
     }
 
+    public static List<TapChangerStepCreationEmbeddable> toEmbeddableRatioTapChangerSteps(List<RatioTapChangerStepInfos> tapChangerStepsInfos) {
+        List<TapChangerStepCreationEmbeddable> ratioTapChangerSteps = new ArrayList<>();
+        if (tapChangerStepsInfos != null) {
+            for (RatioTapChangerStepInfos ratioTapChangerStepInfos : tapChangerStepsInfos) {
+                ratioTapChangerSteps.add(new TapChangerStepCreationEmbeddable(TapChangerType.RATIO, ratioTapChangerStepInfos.getIndex(), ratioTapChangerStepInfos.getRho(), ratioTapChangerStepInfos.getR(), ratioTapChangerStepInfos.getX(), ratioTapChangerStepInfos.getG(), ratioTapChangerStepInfos.getB(), null));
```

Comment:
You could factorize a bit this code, see GeneratorCreationEntity::toEmbeddablePoints

Reply:
✓